jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/applications/ServerlessExplorerNodes.kt

```diff
@@ -25,23 +33,50 @@ class ServerlessApplicationsNode(project: Project) : AwsExplorerPageableNode<Str
             paginationToken?.let { request.nextToken(it) }
         }
 
-        val nodes = response.stacks().filterNotNull().filter { stack ->
-            client.describeStackResources { it.stackName(stack.stackName()) }.stackResources().any { it.resourceType() == LAMBDA_FUNCTION_TYPE }
-        }.map {
-            ServerlessApplicationNode(nodeProject, it.stackName())
-        }
+        val nodes = response.stacks().filterNotNull().asSequence()
+            .filter { !(it.stackStatus() in DELETING_STACK_STATES) }
+            .map { stack -> stack to functionsForStack(stack) }
+            .filter { (_, functions) -> functions.isNotEmpty() }
+            .map { (stack, functions) ->
+                ServerlessApplicationNode(nodeProject, stack.stackName(), functions.map { it.physicalResourceId() })
+            }.toList()
 
         return nodes + paginationNodeIfRequired(response.nextToken())
     }
 
+    private fun functionsForStack(stack: Stack) =
+        client.describeStackResources { it.stackName(stack.stackName()) }
+            .stackResources()
+            .filter { it.resourceType() == LAMBDA_FUNCTION_TYPE && it.resourceStatus() in COMPLETE_RESOURCE_STATES }
+
     private fun paginationNodeIfRequired(nextToken: String?) = when {
         nextToken != null -> listOf(AwsTruncatedResultNode(this, nextToken))
         else -> emptyList()
     }
+
+    private companion object {
+        val COMPLETE_RESOURCE_STATES = setOf(ResourceStatus.CREATE_COMPLETE, ResourceStatus.UPDATE_COMPLETE)
+        val DELETING_STACK_STATES = setOf(StackStatus.DELETE_IN_PROGRESS, StackStatus.DELETE_COMPLETE)
```

Comment:
If the deletion failed, the status of DELETE_IN_PROGRESS would be changed to DELETE_FAILED. This gives customers a weird experience that a stack would be shown again after the status change.

Should we show the stacks in DELETE_IN_PROGRESS as well?

Reply:
I think showing `DELETE_FAILED` is fine - the stack is still there - it just might not be in a 'clean' state. Long term I'd love to find a way of representing the status at the top level (red text maybe?).

Reply:
I would say grey small text like how the Module says the location when you hover next to the node in grey

Reply:
have opened #432 to track this - other changes have been fixed.